widget/hyperlink.go

```diff
@@ -0,0 +1,140 @@
+package widget
+
+import (
+	"image/color"
+	"net/url"
+	"os/exec"
+	"runtime"
+
+	"fyne.io/fyne"
+	"fyne.io/fyne/theme"
+)
+
+// Hyperlink widget is a text component with appropriate padding and layout.
+// When clicked, the default web browser should open with a URL
+type Hyperlink struct {
+	textProvider
+	Text      string
+	Url       string
+	Alignment fyne.TextAlign // The alignment of the Text
+	TextStyle fyne.TextStyle // The style of the hyperlink text
+}
+
+// NewHyperlink creates a new layout widget with the set text content
+func NewHyperlink(text string, sUrl string) *Hyperlink {
+	// ignore the error here, we'll continue anyway
+	return NewHyperlinkWithStyle(text, sUrl, fyne.TextAlignLeading, fyne.TextStyle{})
+}
+
+// NewHyperlinkWithStyle creates a new layout widget with the set text content
+func NewHyperlinkWithStyle(text string, sUrl string, alignment fyne.TextAlign, style fyne.TextStyle) *Hyperlink {
+	hl := &Hyperlink{
+		Text:      text,
+		Url:       sUrl,
+		Alignment: alignment,
+		TextStyle: style,
+	}
+
+	return hl
+}
+
+// SetText sets the text of the hyperlink
+func (hl *Hyperlink) SetText(text string) {
+	hl.Text = text
+	hl.textProvider.SetText(text)
+	Renderer(hl).Refresh()
+}
+
+// SetUrl sets the URL of the hyperlink, taking in a string type
+func (hl *Hyperlink) SetUrl(sUrl string) {
+	hl.Url = sUrl
+}
+
+// SetUrl sets the URL of the hyperlink, taking in a Url type
+func (hl *Hyperlink) SetUrlFromUrl(uUrl *url.URL) {
+	hl.Url = uUrl.String()
+}
+
+// textAlign tells the rendering textProvider our alignment
+func (hl *Hyperlink) textAlign() fyne.TextAlign {
+	return hl.Alignment
+}
+
+// textStyle tells the rendering textProvider our style
+func (hl *Hyperlink) textStyle() fyne.TextStyle {
+	return hl.TextStyle
+}
+
+// textColor tells the rendering textProvider our color
+func (hl *Hyperlink) textColor() color.Color {
+	return theme.HyperlinkColor()
+}
+
+// password tells the rendering textProvider if we are a password field
+func (hl *Hyperlink) password() bool {
+	return false
+}
+
+// object returns the root object of the widget so it can be referenced
+func (hl *Hyperlink) object() fyne.Widget {
+	return hl
+}
+
+// OnMouseDown is called when a mouse down event is captured and triggers any change handler
+func (hl *Hyperlink) OnMouseDown(*fyne.MouseEvent) {
+	if hl.Url != "" {
+		open(hl.Url)
```

Comment:
app.OpenURL is your friend here

Reply:
Thanks!!  You're one step ahead of me here :)

Reply:
Changed in 3fdc033528934b7e8ed483474c570a7f82cbcc9a